### DISCUSSION TITLE
Rename Brew cask formula to `ice`

### DISCUSSION THREAD

**Discussion by uncenter** [jordanbaird/Ice#56] [General]
### Description

This may not be an option for whatever reason but it would be great if we could [rename the cask formula](https://docs.brew.sh/Rename-A-Formula) from `jordanbaird-ice` to just `ice`. There isn't an existing cask with that name (though there is a regular formula for it, but I think that doesn't matter as I can see other names that have both a CLI and cask attached).

**Comment by jordanbaird** [comment]
I would prefer it that way, too, but running `brew audit` on the renamed cask fails with the following:

```sh
 - possible duplicate, cask token conflicts with Homebrew core formula: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/i/ice.rb
```

Could you point me to some of the duplicate names you mentioned? Maybe they make some exceptions, or maybe it doesn't matter as much as I'm thinking. We could also try asking the [Homebrew forum](https://github.com/orgs/Homebrew/discussions) to get a definitive answer.

**Comment by fastnlight0** [comment]
How about something simple, such as `ice-mac`, or `ice-menubar`, or something that tells what it does?

**Comment by jordanbaird** [comment]
@fastnlight0 Homebrew's [cask documentation](https://docs.brew.sh/Adding-Software-to-Homebrew#casks) states that the name should either be the app name, or the app name prefixed by the vendor name in the case of duplicates, so we only really have two options...unless I'm reading into it too much.

**Comment by uncenter** [comment]
> I would prefer it that way, too, but running `brew audit` on the renamed cask fails with the following:
> 
> ```shell
>  - possible duplicate, cask token conflicts with Homebrew core formula: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/i/ice.rb
> ```
> 
> Could you point me to some of the duplicate names you mentioned? Maybe they make some exceptions, or maybe it doesn't matter as much as I'm thinking. We could also try asking the [Homebrew forum](https://github.com/orgs/Homebrew/discussions) to get a definitive answer.

Hmmm, here's one I found.

![Screenshot 2024-04-13 at 07 01 22 (Arc)](https://github.com/jordanbaird/Ice/assets/47499684/5cbc9076-6c6a-4a2a-a928-29b1095b7ce4)

**Comment by uncenter** [comment]
> How about something simple, such as `ice-mac`, or `ice-menubar`, or something that tells what it does?

I like `ice-menubar`! No duplicates there either :D

**Comment by jordanbaird** [comment]
I'll ask over on the forums to see what they think. If they say that either one is fine, which one would we prefer? `ice` is nice and succinct, but has the downside of having to use the full ``brew install --cask ice`` syntax. If we use `ice-menubar`, it can be shortened to ``brew install ice-menubar`` (I say "shortened," but it's actually a bit longer haha).

**Comment by jordanbaird** [comment]
There's a chance they'll be against the rename altogether, as it may introduce some friction for users

**Comment by jordanbaird** [comment]
So far no response on the forums. Do we keep waiting, or go ahead with trying to change it?

**Comment by mtran-code** [comment]
It's been a while since this discussion and I'm not sure if there's been any updates, but I wanted to support this change. I think having just `ice` or `ice-menubar` adds a more professionalism than attaching a legal name to the app. It's also just cleaner.

For using `ice` with `--cask`, I think it's safe to assume that people who use homebrew over traditional install methods know the difference and won't be bothered by the flag, nor care about needing it. Installing is usually a one-time thing anyways. People not too savvy with homebrew are also likely copying and pasting the command.

For `ice-menubar` or something similar, you get the added benefit of people getting a hint at the functionality behind the app. For example, for someone looking for a menubar app might `brew search menubar` just to see what comes back. Or if brew announces `New casks available: ice-menubar`, that might catch people's eye over just ice.

Personally I vote `ice`, but I think either would be an improvement and add some perceived legitimacy to the app.

**Comment by alexistrn** [comment]
Hello there, I came to this discussion because being French, I first wanted to download the application via Homebrew, by typing its name "Ice", then different keywords, but I never found it. I even thought it didn't exist on Homebrew. So I downloaded it directly from the site and having loved it, I was even ready to suggest adding it to Homebrew when I came across this discussion. 

I agree with the `ice-menubar` proposal but wouldn't it be possible to do the opposite of the current name (to keep the name of the developer, which can be understood if you don't want to title it ice-menubar ), writing `ice-jordanbaird` ? This would normally make it possible to find it in the propositions directly when you type "Ice" in the first place (which currently was not the case for me). I propose it like this 🤷‍♂️

**Reply by jordanbaird** [reply]
I'm fine with doing `ice` or `ice-menubar`. I just haven't had time to submit a PR to [homebrew-cask](https://github.com/Homebrew/homebrew-cask).

**Comment by austince** [comment]
Was there any movement here? It seems like the `ice-jordanbaird` cask no longer exists either.

```
> 1 ✗ brew install jordanbaird-ice       
Warning: No available formula with the name "jordanbaird-ice".
==> Searching for similarly named formulae and casks...
Error: No formulae or casks found for jordanbaird-ice.
```

**Reply by jordanbaird** [reply]
I haven't pursued it any further yet, as I've mainly been focusing on app features. The cask should definitely still exist. I just tried it on my end and it works.

**Reply by austince** [reply]
Got it, thanks. Seems to be solved with a `brew update`, sorry for the noise!
